packages/web/src/pages/sign-up-page/pages/SignUpPage.tsx

```diff
@@ -34,30 +57,148 @@ type SignUpEmailValues = {
   email: string
 }
 
+const FormSchema = z.object({
+  email: z
+    .string({ required_error: messages.invalidEmail })
+    .regex(EMAIL_REGEX, { message: messages.invalidEmail })
+})
+
 export const SignUpPage = (props: SignUpPageProps) => {
   const { onNext } = props
-
-  const handleSubmit = useCallback(
+  const dispatch = useDispatch()
+  const navigate = useNavigateToPage()
+  const [checkEmailInUse, result] = useEmailInUse()
```

Comment:
`useIsEmailInUse`? or too wordy?

Reply:
Yeah I was actually wondering a bit about general naming conventions with audius-query here as well. 
I almost renamed this myself but I wasn't sure if it was best practice to leave them the same name as the backend method (in this case `audiusBackend.emailInUse()`)